src/test/mir-opt/const-promotion-extern-static/rustc.BAR-promoted[0].ConstProp.after.mir

```diff
@@ -1,5 +1,5 @@
 // MIR for `BAR`
-// source = MirSource { instance: Item(DefId(0:6 ~ const_promotion_extern_static[317d]::BAR[0])), promoted: Some(promoted[0]) }
+// source = BAR::promoted[0]
```

Comment:
Just realized, these are kind of redundant now?

Could the header just be:
```rust
// MIR for `BAR::promoted[0]` after ConstProp
```

Does anything parse these `// ...` lines?

Reply:
no, i'll change it

Reply:
Uhh.... maybe we shouldn't touch this:

> // #41697 -- we use `with_forced_impl_filename_line()` because
// `def_path_str()` would otherwise trigger `type_of`, and this can
// run while we are already attempting to evaluate `type_of`.

Reply:
I mean, it's fine, just keep using `with_forced_impl_filename_line`.
That's what it's there for, to not print `impl` `Ty`/`TraitRef`s.